etna/search/tests/test_forms.py

```diff
@@ -4,93 +4,95 @@
 
 
 class CatalogueSearchFormTest(SimpleTestCase):
-    def test_valid_opening_date_range_values(self):
+    def test_valid_date_range_values(self):
         """tests inputs accross both date fields with various input formats for form validity"""
-        for label, form_data in (
-            (
-                "input end date with empty start date",
-                {
-                    "group": "tna",
-                    "opening_end_date_0": "01",
-                    "opening_end_date_1": "01",
-                    "opening_end_date_2": "2000",
-                },
-            ),
-            (
-                "input start date with empty end date",
-                {
-                    "group": "tna",
-                    "opening_start_date_0": "31",
-                    "opening_start_date_1": "12",
-                    "opening_start_date_2": "1999",
-                },
-            ),
-            (
-                "input start date before end date",
-                {
-                    "group": "tna",
-                    "opening_start_date_0": "31",
-                    "opening_start_date_1": "12",
-                    "opening_start_date_2": "1999",
-                    "opening_end_date_0": "01",
-                    "opening_end_date_1": "01",
-                    "opening_end_date_2": "2000",
-                },
-            ),
-            (
-                "input for day - start date and end date are same",
-                {
-                    "group": "tna",
-                    "opening_start_date_0": "31",
-                    "opening_start_date_1": "12",
-                    "opening_start_date_2": "1999",
-                    "opening_end_date_0": "31",
-                    "opening_end_date_1": "12",
-                    "opening_end_date_2": "1999",
-                },
-            ),
+
+        for from_field_name, to_field_name in (
+            ("opening_start_date", "opening_end_date"),
+            ("covering_date_from", "covering_date_to"),
```

Comment:
Re: https://github.com/nationalarchives/ds-wagtail/pull/991
> Similar thing here. A separate test for each range would be clearer

I know there must be some reason. but why Andy?

Reply:
I think you know that this approach is very different to yours. I could 'parametirize' a separate method and call that in two separate tests, but I figured this was easier to read.